cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -325,15 +337,79 @@ func newDevicePlugin(sysfsDir, devfsDir string, options cliOptions) *devicePlugi
 		dp.policy = nonePolicy
 	}
 
-	if _, err := os.ReadDir(dp.bypathDir); err != nil {
-		klog.Warningf("failed to read by-path dir: %+v", err)
+	if !options.wslScan {
+		if _, err := os.ReadDir(dp.bypathDir); err != nil {
+			klog.Warningf("failed to read by-path dir: %+v", err)
 
-		dp.bypathFound = false
+			dp.bypathFound = false
+		}
 	}
 
 	return dp
 }
 
+func (dp *devicePlugin) healthStatusForCard(cardPath string) string {
+	if dp.levelzeroService == nil {
+		return pluginapi.Healthy
+	}
+
+	link, err := os.Readlink(filepath.Join(cardPath, "device"))
+	if err != nil {
+		klog.Warning("couldn't read device link for", cardPath)
+
+		return pluginapi.Healthy
+	}
+
+	bdfAddr := filepath.Base(link)
+
+	dh, err := dp.levelzeroService.GetDeviceHealth(bdfAddr)
+	// In case of any errors, deem the device healthy
+	if err != nil {
+		klog.Warningf("Device health retrieval failed: %v", err)
+
+		return pluginapi.Healthy
+	}
+
+	// Direct Health indicators
+	healthOk := dh.Memory && dh.Bus && dh.SoC
+
+	klog.V(4).Infof("Health indicators: Memory=%t, Bus=%t, SoC=%t", dh.Memory, dh.Bus, dh.SoC)
+
+	dt, err := dp.levelzeroService.GetDeviceTemperature(bdfAddr)
+	// In case of any errors, deem the device healthy
+	if err != nil {
+		klog.Warningf("Device temperature retrieval failed: %v", err)
+
+		return pluginapi.Healthy
```

Comment:
You might have healthOk = false, yet here you return Healthy. Are you sure you want that?

Reply:
You are correct. In that case the device should be Unhealthy. Actually, if the health indicators say Unhealthy, then it's really no point in reading the temperatures.